### ISSUE TITLE
A few macOS issues

### ISSUE THREAD

**Issue by leonardchallis** [ronniedroid/getnf#16]
Thanks a lot for this great tool! On macOS there seems to be a few issues. Apologies if some of these aren't considered worth fixing or are more to do with my environment.

I tried these in both bash and zsh, in Kitty and Putty.

1) `readarray: command not found` errors (see example output in 3)

2) .local/bin isn't in the path by default - I did notice other people have added it in their dotfiles though so maybe not one worth worrying about too much and/or just popping something in the readme?

3) The output (whether in iTerm or Kitty and whether using bash or zsh) doesn't seem compatible and (colour?) codes are showing. An example:

```
getnf
\e[34mFonts directory exists, good.\e[0m
\e[34mFonts download directory exists, good.\e[0m
/Users/leonard/.local/bin/getnf: line 70: readarray: command not found
/Users/leonard/.local/bin/getnf: line 73: readarray: command not found
\e[31mAll available fonts have been selected.\e[0m
\e[31mNo fonts were selected selected, exiting.\e[0m
```

Cheers

**Comment by nDimensionalSpace** [comment]
I am seeing the same thing . . .  Current machine is running macOS 12.6.3.  It looks like @ronniedroid submitted a bunch of changes to avoid the need for fzf.  In the process he/she added the use of "readarray", which is problematic on macOS for several reasons, including (1) a sufficiently old version of bash, and (2) the fact that its replacement (zsh) doesn't have readarray at all.

I did fix the readarray references, but some other issue are propagating . . .  I will muck around a bit and check back if I can make any progress.

**Comment by nDimensionalSpace** [comment]
I got a super hack version of this working, as follows:

* change the interpreter line to zsh,
* change the readarray lines to something like
`output=("${(f)input}")`,
and
* down below where the script is using `options` as a variable name, change that to something else, as that is a command in zsh.

There are likely still some other issues outstanding . . .  Probably just easier to install by hand on macOS for now.

**Comment by ronniedroid** [comment]
I will try to figure this out next week. Sorry for the inconvenient,  I do not own a Mac.

**Comment by ronniedroid** [comment]
> Thanks a lot for this great tool! On macOS there seems to be a few issues. Apologies if some of these aren't considered worth fixing or are more to do with my environment.
> 
> I tried these in both bash and zsh, in Kitty and Putty.
> 
>     1. `readarray: command not found` errors (see example output in 3)
> 
>     2. .local/bin isn't in the path by default - I did notice other people have added it in their dotfiles though so maybe not one worth worrying about too much and/or just popping something in the readme?
> 
>     3. The output (whether in iTerm or Kitty and whether using bash or zsh) doesn't seem compatible and (colour?) codes are showing. An example:
> 
> 
> ```
> getnf
> \e[34mFonts directory exists, good.\e[0m
> \e[34mFonts download directory exists, good.\e[0m
> /Users/leonard/.local/bin/getnf: line 70: readarray: command not found
> /Users/leonard/.local/bin/getnf: line 73: readarray: command not found
> \e[31mAll available fonts have been selected.\e[0m
> \e[31mNo fonts were selected selected, exiting.\e[0m
> ```
> 
> Cheers

1. this is the issue I need to fix - if nothing works, we can return back to fzf.
2. I will add to the readme that `./local/bin` needs to be in your path if you use the install script (as you can just run the script from anywhere you want too).
3. This I think can be fixed by using `tput` for color

**Comment by ronniedroid** [comment]
Please try #17 and see if it fixes the MacOS issues, I will merge it anyways after some testing because it is a lot saner than the current method.

**Comment by ronniedroid** [comment]
fixed in #17 which is not merged into master
